egs/wsj/s5/steps/limit_num_gpus.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# This script functions as a wrapper of a bash command that uses GPUs.
+#
+# It sets the CUDA_VISIBLE_DEVICES variable so that it limits the number of GPUs
+# used for programs. It is neccesary for running a job on the grid if the job
+# would automatically grabs all resources available on the system, e.g. a
+# TensorFlow program.
+
+num_gpus=1
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
```

Comment:
don't use parse_options.sh for this and don't source path.sh-- too many dependencies.  Manually parse the --num-gpus option.
Check that the number of remaining command-line args after partsing --num-gpus is nonzero, and die with a usage message if not.  make sure the usage message is informative and contains an example.